### PR TITLE
BOM-1485: Django 2.2 upgrade

### PR DESCRIPTION
**Pull request by robrap**
Will pause the pipeline after merge.

BOM-1485

### CODE REVIEW

File: requirements/pins.txt [hunk 1]
```diff
@@ -1,18 +1,33 @@
-pytz==2016.10  # Pinned to preserve the status quo
-urllib3>=1.24.2,<2.0.0  # Pinned because transifex-client==0.13.6 pins it
-edx-rest-api-client<2.0.0 # Because of django incompatibility
+# Version constraints for pip-installation.
```
Comment (robrap) [inline requirements/pins.txt]: **Reviewer Note**: There are no new pins.  I just added a header and spacing and sorted alphabetically.

File: ecommerce/urls.py [hunk 2]
```diff
@@ -64,7 +64,7 @@ def handler403(_, exception):  # pylint: disable=unused-argument
     url(r'^enterprise/', include(('ecommerce.enterprise.urls', 'enterprise'))),
     url(r'^health/$', core_views.health, name='health'),
     url(r'^i18n/', include(('django.conf.urls.i18n'))),
-    url(r'^jsi18n/$', JavaScriptCatalog.as_view(packages=['courses']), name='javascript-catalog'),
+    url(r'^jsi18n/$', JavaScriptCatalog.as_view(), name='javascript-catalog'),
```
Comment (robrap) [inline ecommerce/urls.py]: This fix for BOM-1487 is being moved to this PR: https://github.com/edx/ecommerce/pull/2887

File: ecommerce/core/migrations/0061_auto_20200406_1721.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        migrations.AlterField(
+            model_name='siteconfiguration',
+            name='send_refund_notifications',
+            field=models.BooleanField(blank=True, default=False, verbose_name='Send refund email notification'),
+        ),
+        migrations.AlterField(
+            model_name='user',
+            name='last_name',
+            field=models.CharField(blank=True, max_length=150, verbose_name='last name'),
+        ),
```
Comment (robrap) [inline ecommerce/core/migrations/0061_auto_20200406_1721.py]: FYI: We will need to fake this migration in Production.  This relates to BOM-1500, where we will hold the last_name to only handle up to 30 characters.
```
root@ecommerce:/edx/app/ecommerce/ecommerce# ./manage.py sqlmigrate core 0061_auto_20200406_1721 --settings=ecommerce.settings.devstack
BEGIN;
-- Alter field enable_enrollment_codes on siteconfiguration
-- Alter field enable_microfrontend_for_basket_page on siteconfiguration
-- Alter field enable_partial_program on siteconfiguration
-- Alter field send_refund_notifications on siteconfiguration
-- Alter field last_name on user
ALTER TABLE `ecommerce_user` MODIFY `last_name` varchar(150) NOT NULL;
COMMIT;
```

File: requirements/pins.txt (hunk 1 above)
Comment (robrap) [reply requirements/pins.txt]: Moved to https://github.com/edx/ecommerce/pull/2890

**Review by awais786 — APPROVED** [review]

**Review by timmc-edx — APPROVED** [review]

File: ecommerce/core/migrations/0061_auto_20200407_1725.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# Generated by Django 2.2.12 on 2020-04-07 17:25
```
Comment (feanil) [inline ecommerce/core/migrations/0061_auto_20200407_1725.py:1]: Where is this migration coming from?  It doesn't seem related to the upgrade?

**Review by feanil — COMMENTED** [review]
One question, otherwise looks good.

File: ecommerce/core/migrations/0061_auto_20200407_1725.py (hunk 4 above)
Comment (robrap) [reply ecommerce/core/migrations/0061_auto_20200407_1725.py:1]: It is coming from the upgrade, but I don't know what change in Django is causing this.  I haven't been concerned to look deeper because the migration itself is a no-op:
```
root@ecommerce:/edx/app/ecommerce/ecommerce# ./manage.py sqlmigrate core 0061 --settings=ecommerce.settings.devstack
BEGIN;
-- Alter field enable_enrollment_codes on siteconfiguration
--
-- Alter field enable_microfrontend_for_basket_page on siteconfiguration
--
-- Alter field enable_partial_program on siteconfiguration
--
-- Alter field send_refund_notifications on siteconfiguration
--
COMMIT;
```

**Review by feanil — APPROVED** [review]
